metal/resource_metal_device.go

```diff
@@ -348,7 +350,61 @@ func resourceMetalDevice() *schema.Resource {
 				Default:     false,
 				ForceNew:    false,
 			},
+			"reinstall": {
+				Type:     schema.TypeList,
+				MaxItems: 1,
+				Optional: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"enabled": {
+							Type:        schema.TypeBool,
+							Description: "Whether the device should be reinstalled instead of destroyed",
+							Optional:    true,
+							Default:     false,
+						},
+						"deprovision_fast": {
+							Type:        schema.TypeBool,
+							Description: "Whether the OS disk should written with 0 bytes",
+							Optional:    true,
+							Default:     false,
+						},
+						"persist_data": {
+							Type:        schema.TypeBool,
+							Description: "Whether the non-OS disks should be kept or wiped during reinstall",
+							Optional:    true,
+							Default:     false,
+						},
+					},
+				},
+			},
 		},
+		CustomizeDiff: customdiff.Sequence(
+			customdiff.ForceNewIf("reinstall", func(_ context.Context, d *schema.ResourceDiff, meta interface{}) bool {
```

Comment:
I haven't used ForceNewIf before, but if I understand this, it seems that we should supply each of these fields as the first argument: `userdata`, `customdata`, `operatingsystem`, and only return true if `reinstall` is not enabled.

Changes to the `reinstall` by itself should not prompt any immediate action or override any other diffs detected.

Reply:
With all three fields being processed as a ForceNewIf, I don't know if Sequence or All would provide a better experience (if not necessary).

https://github.com/hashicorp/terraform-plugin-sdk/blob/main/helper/customdiff/compose.go#L11-L46